lib/metrics.py

```diff
@@ -67,9 +67,14 @@ def emit(self, stats):
             self.fallback_emitter.emit(stats)
 
 
-class MetricsEmitterThread(threading.Thread):
+class BaseMetricsEmitterThread(threading.Thread, metaclass=ABCMeta):
+    """
+    This base class contains all boilerplate code needed to emit metrics.
+    Only thing left to implement in subclass is `_selected_stats_to_emit` method.
+    """
+
     def __init__(self, interval, m2ee):
-        super(MetricsEmitterThread, self).__init__()
+        super(BaseMetricsEmitterThread, self).__init__()
```

Comment:
```suggestion
        super().__init__()
```
Should do the trick, since we are using Python3.